### PR TITLE
removed shadowsocks support and added test for unsupported urls

### PR DESCRIPTION
**Pull request by mnbogner**
It doesn't seem as though shadowsocks has provided much value lately so i think it makes sense to remove it.  This was also a good opportunity to add code to filter out unknown urls (I believe they would currently be processed as http urls and fail).

### CODE REVIEW

**Review by vitriolix — APPROVED** [review]
looks solid

**Review by stevenmcdonald — CHANGES_REQUESTED** [review]
Let's hold off on purging shadowsocks until we hear from Dave about their working prototypes

**Comment by mnbogner** [comment]
it looks like we won't be needing these changes after all so i will close this without merging it.

**Pull request closed without merging**
